pkg/params/settings/config.go

```diff
@@ -96,6 +97,7 @@ type Settings struct {
 }
 
 func ConfigToSettings(logger *zap.SugaredLogger, setting *Settings, config map[string]string) error {
+	var mu sync.Mutex
```

Comment:
you will probably need to have a lock at global level and not at func level
calling this func concurrently will create 2 seperate locks

Reply:
> you will probably need to have a lock at global level and not at func level calling this func concurrently will create 2 seperate locks

True
I did that thinking ConfigToSettings was not called concurrently but when rechecked i see its parent function `WatchConfigMapChanges` called concurrently

Updated now